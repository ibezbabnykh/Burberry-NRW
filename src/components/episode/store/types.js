export const EPISODE_DETAILS_REQUEST = 'EPISODE/EPISODE_DETAILS_REQUEST';
export const EPISODE_DETAILS_SUCCESS = 'EPISODE/EPISODE_DETAILS_SUCCESS';
export const EPISODE_DETAILS_FAILURE = 'EPISODE/EPISODE_DETAILS_FAILURE';