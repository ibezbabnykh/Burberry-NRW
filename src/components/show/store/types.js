export const SHOW_DETAILS_REQUEST = 'SHOW/SHOW_DETAILS_REQUEST';
export const SHOW_DETAILS_SUCCESS = 'SHOW/SHOW_DETAILS_SUCCESS';
export const SHOW_DETAILS_FAILURE = 'SHOW/SHOW_DETAILS_FAILURE';

export const SHOW_EPISODE_LIST_REQUEST = 'SHOW/SHOW_EPISODE_LIST_REQUEST';
export const SHOW_EPISODE_LIST_SUCCESS = 'SHOW/SHOW_EPISODE_LIST_SUCCESS';
export const SHOW_EPISODE_LIST_FAILURE = 'SHOW/SHOW_EPISODE_LIST_FAILURE';