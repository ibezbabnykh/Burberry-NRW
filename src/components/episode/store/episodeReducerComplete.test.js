import reducer from './episodeReducer';
import * as types from './types';

const initialState = {
  data: null,
  error: null,
  loading: false
};

const mockResponse = {
  image: {
    medium: 'https://via.placeholder.com/250x140',
    original: 'https://via.placeholder.com/1280x720'
  },
  name: 'episode name',
  summary: 'episode summary'
}

const mockError = 'Something went wrong';

describe('episode reducer', () => {
  it('should handle EPISODE_DETAILS_REQUEST and return state', () => {
    const requestAction = {
      type: types.EPISODE_DETAILS_REQUEST,
      payload: 1
    }

    const result = reducer(initialState, requestAction);
    expect(result).toMatchSnapshot();
  });

  it('should handle EPISODE_DETAILS_SUCCESS and return state', () => {
    const successAction = {
      type: types.EPISODE_DETAILS_SUCCESS,
      payload: mockResponse
    }

    const result = reducer(initialState, successAction);
    expect(result).toMatchSnapshot();
  });

  it('should handle EPISODE_DETAILS_FAILURE and return state', () => {
    const failureAction = {
      type: types.EPISODE_DETAILS_FAILURE,
      payload: mockError
    }

    const result = reducer(initialState, failureAction);
    expect(result).toMatchSnapshot();
  });
});