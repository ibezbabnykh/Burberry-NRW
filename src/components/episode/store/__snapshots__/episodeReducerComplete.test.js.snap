// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`episode reducer should handle EPISODE_DETAILS_FAILURE and return state 1`] = `
Object {
  "data": null,
  "error": "Something went wrong",
  "loading": false,
}
`;

exports[`episode reducer should handle EPISODE_DETAILS_REQUEST and return state 1`] = `
Object {
  "data": null,
  "error": null,
  "loading": true,
}
`;

exports[`episode reducer should handle EPISODE_DETAILS_SUCCESS and return state 1`] = `
Object {
  "data": Object {
    "image": Object {
      "medium": "https://via.placeholder.com/250x140",
      "original": "https://via.placeholder.com/1280x720",
    },
    "name": "episode name",
    "summary": "episode summary",
  },
  "error": null,
  "loading": false,
}
`;
